addons/account/models/account_move.py

```diff
@@ -3183,8 +3183,13 @@ def add_file_data_results(file_data, invoice):
 
                 except RedirectWarning:
                     raise
-                except Exception:
-                    message = _("Error importing attachment '%s' as invoice (decoder=%s)", file_data['filename'], decoder.__name__)
+                except Exception as e:
+                    message = _(
+                        "Error importing attachment '%s' as invoice (decoder=%s): %s",
+                        file_data['filename'],
+                        decoder.__name__,
+                        str(e),
+                    )
                     invoice.sudo().message_post(body=message)
                     _logger.exception(message)
```

Comment:
```suggestion
                except Exception as e:
                    chatter_message = _(
                        "Error importing attachment '%s' as invoice (decoder=%s)."
                        " Check the file itself, or read the application logs"
                        " for more information.",
                        file_data['filename'],
                        decoder
                    )
                    invoice.sudo().message_post(body=chatter_message)

                    _logger.exception(
                        "Error importing attachment '%s' as invoice (decoder=%s): %s",
                         log_message,
                         file_data['filename'],
                         decoder,
                         str(e)
                    )
```
Would this do?

Reply:
Sorry, I made a few edits after post ^^.
Still re-reading, but I hope you get what I mean with the suggestion.

Reply:
@lordkrandel I'm still not convinced by the invoice chatter message. The user doesn't know what a decoder is. Do we really need to log something in that case? :(

Reply:
Look, let's say I'm a user.

- I upload an invoice XML.
- It appears a blank invoices with no lines, no quantities, no errors at all. Blank draft and the invoice attached. Almost like I pressed the "new" button.
- I think I did something wrong, so I retry.
- Another new invoice completely blank.
- I don't know what's wrong, but there's a header on top (totally unrelated ofc but he doesn't know) saying "you have no credits to do OCR"
- I open a ticket about l10n_it_edi and OCR
- Support knows nothing about this. They don't even know that in the logs there might be an import exception. And they ask... PGI about the ticket.

This happened many times.

What is hard to understand from "We need user feedback somewhere"?
It's a concept that in other contexts I think we well cover with UserErrors, ValidationErrors, warnings... We have user feedback when we send, we have all
kinds of warnings and "Rejected" states.
But if the user imports a file, and it goes wrong, it gets a blank invoice.

I don't care if he knows what a decoder is, but somehow he needs to know there was an error somewhere.

Reply:
@lordkrandel OK then if you insist

Reply:
If find this really painful for a user having errors, if he doesn't have access to the logs, he will have to wait for 3 weeks to get an answer from the support... but I see you're not convinced :man_shrugging: